### ISSUE TITLE
Auto sync is not working properly

### ISSUE THREAD

**Issue by SrushtiPatel** [Odoo-mobile/framework#94]
Hello,

I am facing problem when using auto sync. Auto sync feature is working on one side, i.e. if I make any changes on device, it is running properly at given interval. But if I make changes on server auto sync doesn't work. Please if anyone has solution then help me or guide me if I have missed something.

**Comment by shangx** [comment]
Did you manage to fix the auto sync issue ?

**Comment by SrushtiPatel** [comment]
Unfortunately I haven't found solution for auto sync. Though it is working one way i.e. from device to server, I am not able to identify the issue.

**Comment by shangx** [comment]
In our case the sync works fine both ways, but when we try to add one2many fields, it does'nt sync from device to server anymore.

**Comment by Kelbethor** [comment]
If autosync enabled it's android syncmanager who fires syncing tasks. It should work if model is correctly set and directional properties (candeletelocal, canupdateserver... etc)  and sync is enabled on android account settings. Android fires according to available connection a device load... (Even when you're setting periodicsync, it fires 1 min with wifi conn or 10 with mobile conn. At least this is for me)

**Comment by shangx** [comment]
I know the model works , because if I try with all the fields except the one2many, everything works fine. But when I add the one2many field to my model, the data only syncs one way, from server to device.

**Comment by SrushtiPatel** [comment]
@Kelbethor: I agree with what you said, it is android syncmanager who manages sync operations. Moreover in my case sync is fired in one direction properly. That means my model is working fine, then I am not getting that where am I lacking?!

@shangx: If you are not able to sync from device to server, then there could be chances that you have problem with field values. Possibly, you have not set required values.

**Comment by shangx** [comment]
Yes, but i can't find anywhere what values must be stored in the one2many field before passing it to server. And moreover I don't even want to modify that field, I just use it to get more info, so the value is never changed in the field, it's the exact same value that it gets from the server.

If you have any idea what type of values must be in the one2many field before synking with server I would appreciate a hint ;). Because when I get it from server its an com.odoo.core.orm.OO2MRecord object

**Comment by Kelbethor** [comment]
Well...

### First: I'm neither developer nor contributor in the project, 
just trying to help. That is I'm totally wrong or at least not completely right.

### Second: SO, according to source code one2many fields it's yet-to-be-implemented feature:

```java
public enum FieldType {
        Text, Boolean, ManyToOne, Chips, Selection, Date, Time, DateTime, Blob, RelationType;

        public static FieldType getTypeValue(int type_val) {
            switch (type_val) {
                case 0:
                    return FieldType.Text;
                case 1:
                    return FieldType.Boolean;
                case 2:
                    return FieldType.ManyToOne;
                case 3:
                    return FieldType.Chips;
                case 4:
                    return FieldType.Selection;
                case 5:
                    return FieldType.Date;
                case 6:
                    return FieldType.DateTime;
                case 7:
                    return FieldType.Blob;
                case 8:
                    return FieldType.Time;
            }
            return FieldType.Text;
        }
    }
```
So you should handle it without using framework features or implement it.

### Third: what odoo-mobile orm is expecting from a o2m column?:
A List<Integer> with related model ids

### Fourth: I suggest...
If your model is working just fine without defining o2m column just don't do it. Define m2o column on the other model and then just query it at the right place for showing required info in , for example, a listview...

**Comment by shangx** [comment]
@Kelbethor, That is the approach we are trying now, just implementing models without the one2many, but I wanted to avoid the extra queries and just get all data with one sync :(

Thank you very much for your reply :)
